core/src/main/java/me/aurium/beetle/api/task/futures/TaskFuture.java

```diff
@@ -228,6 +230,38 @@ public Executor defaultExecutor() {
         return (TaskFuture<T>) super.exceptionally(fn);
     }
 
+    /**
+     * This does exactly what you think it does. If you have a main thread, don't call this. (Although if you have a main thread,
+     * you should be using {@link SyncTaskerFuture} instead.
+     *
+     * @param unit time unit
+     * @param amount amount of time units to sleep for
+     * @return itself
+     */
+    public TaskFuture<T> thenSleep(TimeUnit unit, long amount) {
+
+        LockSupport.parkUntil(unit.toNanos(amount));
```

Comment:
There are so many things wrong with this line of code I don't even know where to start. LockSupport is not the right way to wait a given amount of time. Read its documentation and the documentation of this method.

Reply:
I'm not using Thread.sleep here